### PR TITLE
Refactoring: clean on fields and parameter passing being possible by the usage of config.

### PR DESCRIPTION
**Pull request by fermadeiral**
This PR contains only refactoring. Most of them are strongly related with config, some directly and some indirectly.
This PR is for issue #471.

### CODE REVIEW

File: repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java [hunk 1]
```diff
@@ -40,25 +42,24 @@
...
 
     public RTScanner(String runId, List<SerializerEngine> engines) {
+        this.config = RepairnatorConfig.getInstance();
         this.engines = engines;
         this.blackListedRepository = new ArrayList<>();
         this.whiteListedRepository = new ArrayList<>();
         this.tempBlackList = new HashMap<>();
         this.buildRunner = new BuildRunner(this);
         this.inspectBuilds = new InspectBuilds(this);
+        this.inspectBuilds.setMaxSubmittedBuilds(this.config.getMaxInspectedBuilds());
```
Comment (surli) [inline repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java]: Actually if the value are available in the config, those setter are useless too: we can do the set directly inside the constructor of the class no?

File: repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java [hunk 2]
```diff
@@ -124,12 +134,12 @@ public void launch() {
             new Thread(this.inspectJobs).start();
             this.running = true;
 
-            if (this.duration != null) {
+            if (this.config.getDuration() != null) {
                 InspectProcessDuration inspectProcessDuration;
                 if (this.endProcessNotifier != null) {
-                    inspectProcessDuration = new InspectProcessDuration(this.duration, this.inspectBuilds, this.inspectJobs, this.buildRunner, this.endProcessNotifier);
+                    inspectProcessDuration = new InspectProcessDuration(this.config.getDuration(), this.inspectBuilds, this.inspectJobs, this.buildRunner, this.endProcessNotifier);
```
Comment (surli) [inline repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java]: and we should be able to remove this parameter too

File: repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java (hunk 1 above)
Comment (fermadeiral) [reply repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java]: Oh, yes!

File: repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java [hunk 3]
```diff
@@ -83,6 +84,15 @@ public InspectJobs getInspectJobs() {
         return inspectJobs;
     }
 
+    public void initBuildRunner() {
+        LOGGER.info("Init build runner");
+        this.buildRunner.setDockerOutputDir(this.config.getLogDirectory());
```
[the remark this answers is not in the document]
Comment (fermadeiral) [reply repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java]: Done.

File: repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java (hunk 2 above)
Comment (fermadeiral) [reply repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/RTScanner.java]: Done.

**Pull request merged**